src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -45,6 +46,9 @@
     private static final String AD_LEGACY_CONFIG_SEP = "#### SERVICE-DISCOVERY ####";
     private static final String AD_CONFIG_TERM = "#### AUTO-DISCOVERY TERM ####";
     private static final String AD_LEGACY_CONFIG_TERM = "#### SERVICE-DISCOVERY TERM ####";
+    private static final int AD_MAX_NAME_LEN = 80;
+    private static final int AD_MAX_MAG_INSTANCES = (int)Math.log10(1000) + 1;
```

Comment:
That's a bit cryptic, I'd go with `4  // 1000 instances ought to be enough for anyone`